src/main/java/org/apache/sling/feature/cpconverter/repoinit/createpath/CreatePathSegmentProcessor.java

```diff
@@ -27,52 +29,97 @@
 import java.io.FileInputStream;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 
 import static org.apache.jackrabbit.vault.util.Constants.DOT_CONTENT_XML;
 
 public class CreatePathSegmentProcessor {
 
-    private CreatePathSegmentProcessor() {
+    private final RepoPath path;
+    private final Collection<VaultPackageAssembler> packageAssemblers;
+    private final CreatePath cp;
+    private boolean foundType = false;
+    private String repositoryPath = "";
+
+    private Map<String,String> primaryTypeMap = new LinkedHashMap<>();
+    private Map<String,List<String>> mixinTypeMap = new LinkedHashMap<>();
+    
+    public CreatePathSegmentProcessor(@NotNull RepoPath path, 
+                                      @NotNull Collection<VaultPackageAssembler> packageAssemblers, 
+                                      @NotNull CreatePath cp) {
+        this.path = path;
+        this.packageAssemblers = packageAssemblers;
+        this.cp = cp;
     }
 
     /**
      * Process segments of a repopath to createpath, checking packageassemblers for existing primaryType definitions.
-     *
-     * @param path
-     * @param packageAssemblers
-     * @param cp
      * @return
      */
-    public static boolean processSegments(@NotNull RepoPath path, @NotNull Collection<VaultPackageAssembler> packageAssemblers, @NotNull CreatePath cp) {
-        String repositoryPath = "";
-        boolean foundType = false;
+    public boolean processSegments() {
         for (final String part : path.getSegments()) {
-            final String platformName = PlatformNameFormat.getPlatformName(part);
-            repositoryPath = repositoryPath.concat(ConverterConstants.SLASH).concat(platformName);
+            repositoryPath = processSegment(part);
+        }
+        return foundType;
+    }
 
-            boolean segmentAdded = false;
-            //loop all package assemblers and check if .content.xml is defined
-            for (VaultPackageAssembler packageAssembler : packageAssemblers) {
-                File currentContent = packageAssembler.getFileEntry(repositoryPath.concat(ConverterConstants.SLASH).concat(DOT_CONTENT_XML));
-                if (currentContent.exists() && currentContent.isFile()) {
-                    //add segment if jcr:primaryType is defined.
-                    segmentAdded = addSegment(cp, part, currentContent);
-                    if (segmentAdded) {
-                        foundType = true;
-                        break;
+    @NotNull
+    private String processSegment(String part) {
```

Comment:
i believe you want 'part' to be notnull... -> add nullable or notnull annotations.